mempool/reactor.go

```diff
@@ -69,15 +70,34 @@ type Reactor struct {
 	peerRoutines map[p2p.NodeID]*tmsync.Closer
 }
 
+type SmState interface {
+	GetBlockMaxBytes() int64
+	GetBlockMaxGas() int64
+	GetValidatorSize() int
+}
+
 // NewReactor returns a reference to a new reactor.
 func NewReactor(
 	logger log.Logger,
 	config *cfg.MempoolConfig,
 	peerMgr PeerManager,
-	mempool *CListMempool,
 	mempoolCh *p2p.Channel,
 	peerUpdates *p2p.PeerUpdates,
-) *Reactor {
+	memplproxy proxy.AppConnMempool,
+	state SmState,
+	memplMetrics *Metrics,
+	LastBlockHeight int64,
+) (*Reactor, *CListMempool) {
+
+	mempool := NewCListMempool(
```

Comment:
would it make sense to provide a `NewReactorWithMempool(..., mempool mempool.Mempool, ....)` function too that allow you to use something different from a CList mempool?

Reply:
We could, but I am more inclined to tie the implementation to the reactor. This would force people to create their own reactors for their mempool implementations, but I would say that is better than trying to generalize a reactor for many use cases.

Reply:
That makes sense. And I agree with a generalized reactor, since my potential implementation of a `mempool.Mempool` will look very much like the one from `mempool.NewReactor`.